Makefile

```diff
@@ -24,9 +24,12 @@ GOLANGCI_LINT := $(BIN_DIR)/golangci-lint
 KUSTOMIZE := $(BIN_DIR)/kustomize
 
 # See pkg/version.go for details
-SOURCE_GIT_COMMIT ?= $(shell git rev-parse --verify 'HEAD^{commit}')
+SOURCE_GIT_COMMIT ?= $(shell git rev-parse --short HEAD)
 BUILD_VERSION ?= $(shell git describe --always --abbrev=40 --dirty)
-export LDFLAGS="-X github.com/metal3-io/baremetal-operator/pkg/version.Raw=${BUILD_VERSION} -X github.com/metal3-io/baremetal-operator/pkg/version.Commit=${SOURCE_GIT_COMMIT}"
+VERSION_URI = "github.com/metal3-io/baremetal-operator/pkg/version"
+export LDFLAGS="-X $(VERSION_URI).Raw=${BUILD_VERSION} \
+                -X $(VERSION_URI).Commit=${SOURCE_GIT_COMMIT} \
+                -X $(VERSION_URI).BuildTime=$(shell date +%Y-%m-%dT%H:%M:%S%z)"
```

Comment:
Is the important thing the time we ran the compiler, or the time associated with the commit that was built?

Reply:
I was thinking about a compile time as you can rebuild same commit multiple times and knowing compile time could differentiate those builds.

Reply:
That makes some sense. I would expect the results to be essentially the same for the same commit, but these variables are evidence that it might not be.